framework/metricsresource/resource.go

```diff
@@ -0,0 +1,188 @@
+package metricsresource
+
+import (
+	"time"
+
+	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+var (
+	// Name is the identifier of the resource.
+	Name = "metrics"
+)
+
+// Config represents the configuration used to create a new metrics resource.
+type Config struct {
+	// Dependencies.
+	Resource framework.Resource
+
+	// Settings.
+
+	// Namespace is the Prometheus namespace used to create new vectors. The user
+	// has to provide unique namespaces and subsystems. If these settings are not
+	// properly configured and reused the registration of the Prometheus vectors
+	// fails with a panic.
+	Namespace string
+	// Subsystem is the Prometheus subsystem used to create new vectors. The user
+	// has to provide unique namespaces and subsystems. If these settings are not
+	// properly configured and reused the registration of the Prometheus vectors
+	// fails with a panic.
+	Subsystem string
```

Comment:
@rossf7 I would like to improve the godoc as soon as we know how to use it based on experience.